src/providers/cmdline/cmdline.go

```diff
@@ -141,3 +128,62 @@ func parseCmdline(cmdline []byte) (url string) {
 
 	return
 }
+
+// getRawConfig gets the raw configuration data from p.configUrl.
+// Supported URL schemes are:
+// http://	remote resource accessed via http
+// oem://	local file in /usr/share/oem
+func (p *provider) getRawConfig() bool {
+	url, err := url.Parse(p.configUrl)
+	if err != nil {
+		p.logger.Err("failed to parse url: %v", err)
+		return false
+	}
+
+	switch url.Scheme {
+	case "http":
+		if resp, err := p.client.Get(p.configUrl); err == nil {
+			defer resp.Body.Close()
+
+			switch resp.StatusCode {
+			case http.StatusOK, http.StatusNoContent:
+			default:
+				p.logger.Debug("failed fetching: HTTP status: %s", resp.Status)
+				return false
+			}
+
+			p.logger.Debug("successfully fetched")
+			if p.rawConfig, err = ioutil.ReadAll(resp.Body); err != nil {
+				p.logger.Err("failed to read body: %v", err)
+				return false
+			}
+		} else {
+			p.logger.Warning("failed fetching: %v", err)
+		}
+	case "oem":
+		path := filepath.Clean(url.Path)
+		if !filepath.IsAbs(path) {
+			p.logger.Err("oem path is not absolute: %q", url.Path)
+			return false
+		}
+
+		getPath := func(which string) string {
```

Comment:
Closures introduce a lot of complexity with their strange scoping rules. There is hardly any benefit (in this case) to using a closure.

All go code looks like a shell script.